apps/coordinator/src/components/Wallet/TransactionsTab/FeeBumping/utils.ts

```diff
@@ -0,0 +1,975 @@
+
+  return formattedFee + (includeSuffix ? " sats" : "");
+};
+
+/**
+ * Extracts global extended public keys from wallet nodes for PSBT inclusion
+ * @param depositNodes - Wallet deposit nodes
+ * @param changeNodes - Wallet change nodes
+ * @returns Array of GlobalXpub objects
+ */
+export const extractGlobalXpubsFromWallet = (
+  depositNodes: any,
+  changeNodes: any,
+): GlobalXpub[] => {
+  const globalXpubs: GlobalXpub[] = [];
+  const seenXpubs = new Set<string>(); // Prevent duplicates
+
+  // Helper to process a single node
+  const processNode = (node: any) => {
+    if (!node?.multisig?.braidDetails) return;
+
+    try {
+      const braidDetails = JSON.parse(node.multisig.braidDetails);
+      const extendedPublicKeys = braidDetails.extendedPublicKeys || [];
+
+      extendedPublicKeys.forEach((epk: any) => {
+        // Use base58String as the unique identifier
+        const xpubString = epk.base58String || epk.xpub;
+        if (!xpubString || seenXpubs.has(xpubString)) return;
+
+        seenXpubs.add(xpubString);
+
+        const globalXpub: GlobalXpub = {
+          xpub: xpubString,
+          masterFingerprint:
+            epk.rootFingerprint || epk.masterFingerprint || "00000000",
+          path: epk.path || "m",
+        };
+
+        globalXpubs.push(globalXpub);
+      });
+    } catch (error) {
+      console.warn("Error parsing braidDetails for globalXpubs:", error);
+    }
+  };
+
+  // Process all deposit nodes
+  if (depositNodes) {
+    Object.values(depositNodes).forEach(processNode);
+  }
+
+  // Process all change nodes
+  if (changeNodes) {
+    Object.values(changeNodes).forEach(processNode);
+  }
+
+  return globalXpubs;
+};
+
+/**
+ * Analyzes a transaction and provides fee bumping recommendations based on
+ * current network fee estimates and transaction characteristics
+ *
+ * @param txHex - The raw transaction hex string
+ * @param fee - The current transaction fee in satoshis
+ * @param network - The Bitcoin network being used
+ * @param availableUtxos - Available UTXOs for fee bumping
+ * @param blockchainClient - The blockchain client for fee estimation
+ * @param walletConfig - Wallet configuration parameters
+ * @returns Fee bumping analysis and recommendations
+ */
+export const analyzeTransaction = async (
+  txHex: string,
+  fee: number,
+  network: Network,
+  availableUtxos: FeeUTXO[],
+  blockchainClient: BlockchainClient,
+  walletConfig: {
+    requiredSigners: number;
+    totalSigners: number;
+    addressType: string;
+  },
+  feePriority: FeePriority = FeePriority.MEDIUM,
+): Promise<FeeBumpRecommendation> => {
+  // Get fee estimates for different confirmation targets
+  const [highPriorityFee, mediumPriorityFee, lowPriorityFee] =
+    await Promise.all([
+      getFeeEstimate(blockchainClient, CONFIRMATION_TARGETS.HIGH),
+      getFeeEstimate(blockchainClient, CONFIRMATION_TARGETS.MEDIUM),
+      getFeeEstimate(blockchainClient, CONFIRMATION_TARGETS.LOW),
+    ]);
```

Comment:
This looks to me like it should probably be converted into a tanstack query hook. what's crazy too is that each of these makes a network call but in some clients you can get each of these with a single call in some cases. Either then this util should be converted into a hook to use the new query hook OR the function should just take in an object of feeEstimates to work with.